cocos/core/scheduler.ts

```diff
@@ -749,6 +750,33 @@ export class Scheduler extends System {
         }
     }
 
+    /**
+     * @en Unschedule a timer. It is invoked by CallbackTimer when a timer is finished.
+     * @param timerToUnschedule The timer to be unscheduled.
+     * @param target The target of the timer.
+     * @engineInternal
+     */
+    public unscheduleForTimer (timerToUnschedule: CallbackTimer, target: ISchedulable): void {
+        const targetId = (target.uuid || target.id) as string;
+        const element = this._hashForTimers[targetId];
+        const timers = element.timers;
+        if (!timers) {
+            return;
+        }
+
+        for (let i = 0, li = timers.length; i < li; i++) {
+            const timer = timers[i];
+            if (timer === timerToUnschedule) {
+                timers.splice(i, 1);

```

Comment:
Doesn't it need to `CallbackTimer.put(timer);` after `timers.splice(i, 1)`?
I saw there was the logic of re-using the timer instance in `unschedule` function.